### ISSUE TITLE
Does writing logs from multiple processes to the same log file at the same time cause problems

### ISSUE THREAD

**Issue by liuhao123t** [SergiusTheBest/plog#297]
Does writing logs from multiple processes to the same log file at the same time cause problems

**Comment by SergiusTheBest** [comment]
Writing to the same file from multiple processes is atomic and cause no issues. 

What's problematic is log rotation as it requires inter-process synchronization. Also different processes may have different log rotation parameters. So if you do not use log rotation everything will be fine.

**Comment by liuhao123t** [comment]
> Writing to the same file from multiple processes is atomic and cause no issues.
> 
> What's problematic is log rotation as it requires inter-process synchronization. Also different processes may have different log rotation parameters. So if you do not use log rotation everything will be fine.

What if I set the same  log rotation  parameters for all my processes，one process write fast ，the other one write slow, will  they rotate at the same time?

**Comment by SergiusTheBest** [comment]
@liuhao123t No, it won't work as there is no inter-process synchronization. So they may start rotating at the same time, or one after another.

**Comment by Chris55** [comment]
> Writing to the same file from multiple processes is atomic and cause no issues.
> 
> What's problematic is log rotation as it requires inter-process synchronization. Also different processes may have different log rotation parameters. So if you do not use log rotation everything will be fine.

How to disable log rotation?

**Comment by SergiusTheBest** [comment]
> How to disable log rotation?

It's disabled by default. Unless you set log files number and log file max size.
